tests/Unit/Domain/FunctionsOfTime/Test_IntegratedFunctionOfTime.cpp

```diff
@@ -0,0 +1,170 @@
+  }
+}
+
+void test_out_of_order_update() {
+  const std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime> func(
+      std::make_unique<domain::FunctionsOfTime::IntegratedFunctionOfTime>(
+          0.0, std::array<double, 2>{0., 0.5}, 0.5, false));
+
+  func->update(2.5, {{2.5, 3.}}, 3.0);
+  func->update(1.5, {{1.5, 2.}}, 2.0);
+
+  CHECK_THROWS_WITH(
+      func->func(0.7),
+      Catch::Matchers::ContainsSubstring("Attempt to evaluate at time") and
+          Catch::Matchers::ContainsSubstring(
+              ", which is after the expiration time 5"));
+  CHECK_THROWS_WITH(
+      func->func(1.3),
+      Catch::Matchers::ContainsSubstring("Attempt to evaluate at time 1.30") and
+          Catch::Matchers::ContainsSubstring(
+              ", which is after the expiration time 5"));
+  CHECK_THROWS_WITH(
+      func->func(1.9),
+      Catch::Matchers::ContainsSubstring("Attempt to evaluate at time") and
+          Catch::Matchers::ContainsSubstring(
+              ", which is after the expiration time 5"));
+
+  func->update(0.5, {{0.5, 1.0}}, 1.0);
+
+  // This one should pass now
+  CHECK(func->func_and_deriv(0.7) ==
+        std::array<DataVector, 2>{DataVector{0.5}, DataVector{1.}});
+  CHECK_THROWS_WITH(
+      func->func(1.3),
+      Catch::Matchers::ContainsSubstring("Attempt to evaluate at time 1.30") and
+          Catch::Matchers::ContainsSubstring(
+              ", which is after the expiration time 1.0"));
+  CHECK_THROWS_WITH(
+      func->func(1.9),
+      Catch::Matchers::ContainsSubstring("Attempt to evaluate at time") and
+          Catch::Matchers::ContainsSubstring(
+              ", which is after the expiration time 1.0"));
+
+  func->update(2.0, {{2., 2.5}}, 2.5);
+
+  CHECK_THROWS_WITH(
+      func->func(1.3),
+      Catch::Matchers::ContainsSubstring("Attempt to evaluate at time 1.30") and
+          Catch::Matchers::ContainsSubstring(
+              ", which is after the expiration time 1.0"));
+  CHECK_THROWS_WITH(
+      func->func(1.9),
+      Catch::Matchers::ContainsSubstring("Attempt to evaluate at time") and
+          Catch::Matchers::ContainsSubstring(
+              ", which is after the expiration time 1.0"));
+
+  func->update(1.0, {{1., 1.5}}, 1.5);
+
+  CHECK(func->func_and_deriv(1.3) ==
+        std::array<DataVector, 2>{DataVector{1.}, DataVector{1.5}});
+  CHECK(func->func_and_deriv(1.9) ==
+        std::array<DataVector, 2>{DataVector{1.5}, DataVector{2.}});
+  CHECK(func->func_and_deriv(2.2) ==
+        std::array<DataVector, 2>{DataVector{2.}, DataVector{2.5}});
+  CHECK(func->func_and_deriv(2.8) ==
+        std::array<DataVector, 2>{DataVector{2.5}, DataVector{3.}});
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.FunctionsOfTime.IntegratedFunctionOfTime",
+                  "[Domain][Unit]") {
+  FunctionsOfTime::register_derived_with_charm();
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<double> dist(-10., 10.);
+  const DataVector used_for_size(300);
```

Comment:
Use a `size_t` rather than an uninitialized `DataVector`.  The `make_with_random_values` function will accept either.